src/services/storage/drivers/relational/index.ts

```diff
@@ -91,11 +110,38 @@ export class RelationalStorage implements StorageDriver {
 			if (!url) throw NotFoundError()
 			return storedUrl
 		}
-		public async save(url: string): Promise<StoredUrl> {
-			const [storedUrl] = await this.storage.db.table<StoredUrl>('urls').insert({ url }).returning('*')
+
+		public async save(urlBody: UrlRequestData): Promise<StoredUrl> {
+			const url = urlBody.url
+			const urlInfo: Partial<UrlWithInformation> = {
```

Comment:
The actual type here should be UrlInformation (which does not exist).

Maybe we should sort the types like this?

```typescript
interface Url {
    id: string
    // ...
}
interface UrlInformation {
    ip: string
    // ...
}

type UrlWithInformation = Url & UrlInformation

```

Reply:
don't you think Partial is good enough?  I can do the refactor but it will take me a while

Reply:
It should be pretty simple actually. If that's too much time, just leave it as is.